openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,214 @@
+import os
+import copy
+
+import pyblish.api
+
+from openpype.pipeline import publish
+from openpype.lib import (
+
+    is_oiio_supported,
+)
+
+from openpype.lib.transcoding import (
+    convert_colorspace,
+    get_transcode_temp_directory,
+)
+
+from openpype.lib.profiles_filtering import filter_profiles
+
+
+class ExtractColorTranscode(publish.Extractor):
+    """
+    Extractor to convert colors from one colorspace to different.
+
+    Expects "colorspaceData" on representation. This dictionary is collected
+    previously and denotes that representation files should be converted.
+    This dict contains source colorspace information, collected by hosts.
+
+    Target colorspace is selected by profiles in the Settings, based on:
+    - families
+    - host
+    - task types
+    - task names
+    - subset names
+    """
+
+    label = "Transcode color spaces"
+    order = pyblish.api.ExtractorOrder + 0.019
+
+    optional = True
+
+    # Supported extensions
+    supported_exts = ["exr", "jpg", "jpeg", "png", "dpx"]
+
+    # Configurable by Settings
+    profiles = None
+    options = None
+
+    def process(self, instance):
+        if not self.profiles:
+            self.log.debug("No profiles present for color transcode")
+            return
+
+        if "representations" not in instance.data:
+            self.log.debug("No representations, skipping.")
+            return
+
+        if not is_oiio_supported():
+            self.log.warning("OIIO not supported, no transcoding possible.")
+            return
+
+        profile = self._get_profile(instance)
+        if not profile:
+            return
+
+        repres = instance.data.get("representations") or []
+        for idx, repre in enumerate(list(repres)):
+            self.log.debug("repre ({}): `{}`".format(idx + 1, repre["name"]))
+            if not self._repre_is_valid(repre):
+                continue
+
+            colorspace_data = repre["colorspaceData"]
+            source_colorspace = colorspace_data["colorspace"]
+            config_path = colorspace_data.get("configData", {}).get("path")
+            if not os.path.exists(config_path):
+                self.log.warning("Config file doesn't exist, skipping")
+                continue
+
+            repre = self._handle_original_repre(repre, profile)
```

Comment:
This needs to come after copying the tags from original representation to the new representation, else the new representation will inherit the potential `delete` tag and no representations will be available for integration.

Reply:
This has not been implemented.